cmake/modules/BuildBoost.cmake

```diff
@@ -142,7 +142,7 @@ function(do_build_boost root_dir version)
     #"--buildid=ceph" # changes lib names--can omit for static
     ${boost_features})
   set(install_command
-    ${b2} install)
+    ${b2} ${boost_features} install)
```

Comment:
i don't quite understand this change. could you elaborate on this a little bit?

Reply:
me neither... but it seems like that if the "build step" fails, and I run cmake again with different flags, they are taken into effect only in the "install step" (where boost is also being rebuilt due to the initial failure)

Reply:
me neither... but it seems like the 'install step" also tries to build boost, and fails with the same errors

Reply:
i think a better fix is https://github.com/ceph/ceph/pull/49403

Reply:
ok. makes sense.
but won't it still be useful to preserve user supplied `BOOST_CXXFLAGS`?

Reply:
no. on the contrary, i think the less knobs the better. the whole point of having `BOOST_CXXFLAGS` was to collect the options in a more readable way. it is not supposed to be an option.